F#98 modular xacros

guihomework opened the pull request:
## Proposed changes

Fixes #98 
Additional tests were created to prove backward compatibility of the new code to old hand instantiation (in case people have their own somewhere) to not break things. Almost all files in robots folder can be instantiated with the new macro (the one that cannot because they have too specific code inside the instantiation xacro were removed), and of course new instantiation were provided for all hands anyway.

All test pass and all visual test pass as well. Not all hands have been added to the visual test, but the one existing pass. 2 more were added to ensure lite and extra lite are really visible.

This code was tested on our hardware (standard 5 finger motor hand)

Finally, we cannot provide this for noetic (no machine on noetic yet here) but all deprecated warnings were already removed as well within the same PR.

## Checklist

If the task is applicable to this pull request (see applicability criteria in brackets), make sure it is completed before checking the corresponding box. Otherwise, tick the box right away. Make sure that **ALL** boxes are checked **BEFORE** the PR is merged.

- [X] Manually tested that added code works as intended (if any functional/runnable code is added).
- [X] Added automated tests (if a new class is added (Python or C++), interface of that class must be unit tested).
- [X] Tested on real hardware (if the changed or added code is used with the real hardware).
- [X] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc.)

rhaschke committed:
Replace $(find xacro)/xacro -> xacro

rhaschke committed:
Remove deprecated --inorder option

rhaschke committed:
Fix missing xacro namespace prefixes

guihomework committed:
Added test of backward compatibility

guihomework committed:
Fixed a 7 years old typo introduced when upgrading to indigo

guihomework committed:
Fixed race-condition on tests using the same file

Additionally modernized the test to output found files
(following pr2_description upstream)

guihomework committed:
Improved xacro for modularity and to permit sensor selection

Use modern xacro to simplify URDF generation for more hands and more
types of sensors. Also permits to deactivate gazebo sensors
Maintained backward compatibility of old parameters
Use python pi

guihomework committed:
Fixed call to deprecated packages in hand tests

guihomework commented:
I should add that I replaced all M_PI as well, without even seeing #107 but fixing it :-)

lianghongzhuo commented:
Thanks for your work :) 
In noetic version, M_PI was removed but added pi instead. Besides that, I didn't see much difference between noetic and melodic: [link](https://github.com/shadow-robot/sr_common/compare/melodic-devel...noetic-devel)

guihomework commented:
Hi @lianghongzhuo , this PR is for melodic, and removes M_PI usage within most xacros
https://github.com/ubi-agni/sr_common/commit/9ed52f1dced2a31b733d03f01e87146f4d0348cf#diff-379946d558cf8c28df553a8b3e1e896959353b3c92adbd9236a51ac1932f6953L10
then a backward compatible M_PI definition is added as a transition here
https://github.com/ubi-agni/sr_common/commit/9ed52f1dced2a31b733d03f01e87146f4d0348cf#diff-f2c8470e92e906d97489c0ba424c35591a640d2d934a74b37be4d672aa3b686dR44

I think the Shadow Team should then decide if they want to keep or not the backward compatibility in the migration to noetic. I think it is important to not break melodic and offer this deprecated at first.

guihomework commented:
> Thanks for your work :)
> In noetic version, M_PI was removed but added pi instead. Besides that, I didn't see much difference between noetic and melodic: [link](https://github.com/shadow-robot/sr_common/compare/melodic-devel...noetic-devel)

actually you are right there have been some changes in noetic (here https://github.com/shadow-robot/sr_common/pull/102/commits/471acfc8001261a8483dc4f064bdaea1bf088271) that is related to your request, and I personally don't think the solution of redefining pi is good as pi is already defined within python with xacro.

guihomework commented:
@beatrizleon no need for melodic, we need to migrate to noetic in 3 months or so, and wait partially for you. We would like to have those changes in noetic. We have been running our code with this change for quite a while in melodic. 
I just rebased all on top of your latest melodic recently and proposed the refactor part (not our tactile stuff) to you for melodic because we still run melodic now, and all the features of xacro permitting the refactor existed in melodic already.

lianghongzhuo commented:
@guihomework so maybe you could change this pull request to target to noetic-devel?

guihomework commented:
@lianghongzhuo I fear that @beatrizleon will not use it as is anyway. I wait for her advice. I don't have a way to test on real hardware though, since I am waiting for Shadow to migrate. It is a chicken egg problem. But I know that the change work on melodic, on our hardware

beatrizleon commented:
@guihomework thanks for clarifying. @lianghongzhuo 
 Yes, I will create the same in noetic and add a few more changes that we require. I am also refactoring the mesh folder so it will be a big refactoring. Hopefully doing it next week.

guihomework committed:
Fixed broken backward_compatibility

variable name was forgotten in the backward compat file, also hiding
typo mistakes from the xacro expansion due to lazy evaluation

guihomework commented:
replaced by #120
